webClient/src/app/shared/editor-control/editor-control.service.ts

```diff
@@ -20,18 +20,47 @@ import { UtilsService } from '../utils.service';
 import { HttpService } from '../http/http.service';
 import { SnackBarService } from '../snack-bar.service';
 import { Observer } from 'rxjs/Observer';
-import { Http } from '@angular/http';
+import { Http, Headers } from '@angular/http';
 import * as _ from 'lodash';
 import { MatDialog } from '@angular/material';
 import { Angular2InjectionTokens } from 'pluginlib/inject-resources';
 import { MessageDuration } from "../message-duration";
+import { OverwriteDatasetComponent } from '../../shared/dialog/overwrite-dataset/overwrite-dataset.component';
 import * as monaco from 'monaco-editor'
 
 let stateCache = {};
 let lastFile;
+const WHITESPACE_256 = String.fromCharCode.apply(null, new Array(256).fill(0x20));
```

Comment:
I think you can use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/repeat which makes me think is more performant than creating an array with size 256 and then converting it back to a string